idp/provider.go

```diff
@@ -86,6 +86,8 @@ func GetIdProvider(typ string, subType string, clientId string, clientSecret str
 		return NewCasdoorIdProvider(clientId, clientSecret, redirectUrl, hostUrl)
 	} else if isGothSupport(typ) {
 		return NewGothIdProvider(typ, clientId, clientSecret, redirectUrl)
+	} else if typ == "Bilibili" {
+		return NewBilibiliIdProvider(clientId, clientSecret, redirectUrl)
```

Comment:
@jiak94 thanks for this PR! BTW, the frontend JS code also needs changes to allow users to choose this new provider, see previous PRs about how to do it: https://github.com/casdoor/casdoor/pulls?q=is%3Apr+is%3Aclosed+provider

Reply:
will update the PR later this week.

Reply:
@jiak94 hi, any update?